backend/src/graphql/resolver/UserResolver.ts

```diff
@@ -729,27 +737,35 @@ export class UserResolver {
     if (moderator.id === userId) {
       throw new LogError('Administrator can not change his own role')
     }
-    // change isAdmin
-    switch (user.isAdmin) {
-      case null:
-        if (isAdmin) {
-          user.isAdmin = new Date()
-        } else {
-          throw new LogError('User is already an usual user')
-        }
-        break
-      default:
-        if (!isAdmin) {
-          user.isAdmin = null
-        } else {
-          throw new LogError('User is already admin')
-        }
-        break
+    // if user role(s) should be deleted by role=null as parameter
+    if (role === null && user.userRoles) {
+      if (user.userRoles.length > 0) {
+        // remove all roles of the user
+        await UserRole.delete({ userId: user.id })
+        user.userRoles.length = 0
+      } else if (user.userRoles.length === 0) {
+        throw new LogError('User is already an usual user')
+      }
+    } else if (isUserInRole(user, role)) {
+      throw new LogError('User already has role=', role)
+    }
+    // if role shoud be set
+    if (role) {
+      if (user.userRoles === undefined) {
+        user.userRoles = [] as UserRole[]
+      }
+      if (user.userRoles.length < 1) {
+        user.userRoles.push(UserRole.create())
+      }
+      user.userRoles[0].createdAt = new Date()
+      user.userRoles[0].role = role
+      user.userRoles[0].userId = user.id
+      await UserRole.save(user.userRoles[0])
     }
-    await user.save()
+    // await user.save()
```

Comment:
yes, because it works and the only modification was done on userRole and not on user. the afterwards reload of the user at the end give the real stored data.